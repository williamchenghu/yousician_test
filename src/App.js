import React, { Component } from 'react'
import SongsListContainer from './containers/SongsListContainer'
//search bar
import SearchBar from 'material-ui-search-bar'
import Slider from 'material-ui/Slider'

export class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      msg: '',
      searchMsg: '',
      filterLevel: 15,
    }
  }

  render() {
    let { searchMsg, filterLevel } = this.state
    return (
      <div>
        <SearchBar
          id="searchBar"
          onChange={msg => {
            this.setState({ searchMsg: msg || '' })
          }}
          onRequestSearch={() => {}}
        />
        <div className="silder">
          <span>Yousician Level: 1 to {filterLevel}</span>
          <Slider
            id="silder"
            min={1}
            max={15}
            step={1}
            sliderStyle={{
              margin: '1em 0em',
            }}
            value={filterLevel}
            onChange={(e, value) => {
              this.setState({ filterLevel: value })
            }}
          />
        </div>

        <SongsListContainer searchMsg={searchMsg} filterLevel={filterLevel} />
      </div>
    )
  }
}

export default App
