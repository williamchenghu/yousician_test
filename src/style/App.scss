$y-green: #4fc514;
$y-grey: #dbeaff;
$y-dark-grey: #878789;
$y-black: #0e0f13;

body {
  margin: 1em 5em;
  padding: 0;
  font-family: sans-serif;
  background: $y-black;
  color: $y-grey;
}

.logo {
  background: $y-green;
  width: 4em;
  padding: 1em;
  border-radius: 1em;
}
.card {
  display: block;
  .grid {
    display: inline-block;
    vertical-align: middle;
    margin-right: 1em;
  }
  .settings {
    float: right;
  }
}
.padding-space {
  padding: 1em 2em;
}
.donut-level {
  .donut {
    max-width: 3em;
  }
  .number {
    position: relative;
    text-align: center;
    top: -2em;
  }
}
