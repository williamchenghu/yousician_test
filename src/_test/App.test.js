import React from 'react'
import App from '../App'

describe('App', () => {
  describe('activate: App', () => {
    it('should be defined', () => {
      // the component should be define
      expect(App).toBeDefined()
    })

    it('should render correctly', () => {
      //prepare
      let wrapper = shallow(<App />)
      //check
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('actions: App', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(<App />)
    })
    afterEach(() => {
      jest.clearAllMocks()
    })

    it('should filterLevel be changed when silder move', () => {
      // the component should be define
      wrapper.find('#silder').simulate('change', {}, 5)
      expect(wrapper.state().filterLevel).toEqual(5)
    })

    it('should listen to the OnChange of search bar', () => {
      // the component should be define
      wrapper.find('#searchBar').simulate('change', 'test')
      expect(wrapper.state().searchMsg).toEqual('test')
    })
  })
})
