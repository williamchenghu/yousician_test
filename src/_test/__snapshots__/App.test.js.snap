// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App activate: App should render correctly 1`] = `
<div>
  <SearchBar
    closeIcon={
      <NavigationClose
        color="#9e9e9e"
      />
    }
    dataSource={Array []}
    dataSourceConfig={
      Object {
        "text": "text",
        "value": "value",
      }
    }
    disabled={false}
    hintText="Search"
    id="searchBar"
    onChange={[Function]}
    onRequestSearch={[Function]}
    searchIcon={
      <ActionSearch
        color="#9e9e9e"
      />
    }
    spellCheck={false}
    value=""
  />
  <div
    className="silder"
  >
    <span>
      Yousician Level: 1 to 
      15
    </span>
    <Slider
      axis="x"
      disableFocusRipple={false}
      disabled={false}
      id="silder"
      max={15}
      min={1}
      onChange={[Function]}
      required={true}
      sliderStyle={
        Object {
          "margin": "1em 0em",
        }
      }
      step={1}
      style={Object {}}
      value={15}
    />
  </div>
  <SongsListContainer
    filterLevel={15}
    searchMsg=""
  />
</div>
`;
